tests/src/core/testqgslegendrenderer.cpp

```diff
@@ -1435,6 +1437,78 @@ void TestQgsLegendRenderer::testDiagramAttributeLegend()
   QgsProject::instance()->removeMapLayer( vl4 );
 }
 
+void TestQgsLegendRenderer::testDiagramMeshLegend()
+{
+  const QString uri_1( QString( TEST_DATA_DIR ) + QStringLiteral( "/mesh/mesh_z_ws_d_vel.nc" ) ); //mesh with dataset group "Bed Elevation", "Water Level", "Depth" and "Velocity"
+
+  QTemporaryDir tempDir;
+  const QString uri( tempDir.filePath( QStringLiteral( "mesh.nc" ) ) );
+
+  QFile::copy( uri_1, uri );
+  QgsMeshLayer *layer = new QgsMeshLayer( uri, QStringLiteral( "mesh" ), QStringLiteral( "mdal" ) );
+  QVERIFY( layer->isValid() );
+  QCOMPARE( layer->datasetGroupCount(), 4 );
+
+  QgsProject::instance()->addMapLayer( layer );
+
+  int scalarIndex = 0;
+  int vectorIndex = -1;
+
+  QgsMeshRendererSettings rendererSettings = layer->rendererSettings();
+  rendererSettings.setActiveScalarDatasetGroup( scalarIndex );
+  rendererSettings.setActiveVectorDatasetGroup( vectorIndex );
+  layer->setRendererSettings( rendererSettings );
+
+  std::unique_ptr< QgsLayerTree > root( std::make_unique<QgsLayerTree>() );
+  root->addLayer( layer );
+  std::unique_ptr<QgsLayerTreeModel> legendModel( std::make_unique<QgsLayerTreeModel>( root.get() ) );
+
+  QgsLegendSettings settings;
+
```

Comment:
You need to call _setStandardTestFont here to normalise the legend's font, or the test won't work reliably across different platforms

Reply:
That was the solution that fixed the issue! 👍